### ISSUE TITLE
Publish-NavContainerApp -upgrade -install fails if app was not installed before

### ISSUE THREAD

**Issue by jeesus450** [microsoft/navcontainerhelper#2286]
Looks like it is related to the change in the following issue -> (https://github.com/microsoft/navcontainerhelper/pull/2093)

**Describe the issue**
Script I executed:
```
$containerName = "test"
$credentials = Get-Credential
$scope = "Global"
$appFile = "<somepath>\<filename>.app"
$parameters = @{
    containerName = $containerName
    credential = $credentials
    sync = $true
    ignoreIfAppExists = $true
    install = $true
    upgrade = $true
    skipVerification = $true
}

if ($scope.Length -eq 0 -or $scope -eq "Global") {
    $parameters.Add("scope","Global")
}
elseif ($scope -eq "Tenant") {
    $parameters.Add("scope","Tenant")
    $parameters.Add("tenant","default")
}
Publish-BcContainerApp @parameters -appFile $appFile
```
I have a new problem with this code (change from #2093) when I install a new app:

```
if($navAppInfoFromDb.ExtensionDataVersion -eq  $navAppInfoFromDb.Version){
        $upgrade = $false
+   } else {
+       $install = $false
    }
```

If -install and -upgrade both are specified and there was no previous version of the app ( what means install), then after the sync $navAppInfoFromDb.ExtensionDataVersion = $null and does not equal $navAppInfoFromDb.Version what means $install = $false what is wrong.

I had to do 2 changes to this to make it work:
1. add extra condition to the code block mentioned:  -or ($navAppInfoFromDb.ExtensionDataVersion.Length -eq 0)
```
if(($navAppInfoFromDb.ExtensionDataVersion -eq  $navAppInfoFromDb.Version) -or `
($navAppInfoFromDb.ExtensionDataVersion.Length -eq 0))
{
     $upgrade = $false
} else {
     $install = $false
}
```

2. and I had to add $upgrade = $false under if($ignoreIfAppExists) {
} <- this was needed, because if the app was intalled before, then the code under point 1 was not executed at all.
```
$appInfo = (Get-NAVAppInfo -ServerInstance $serverInstance -Name $navAppInfo.Name -Publisher $navAppInfo.Publisher -Version $navAppInfo.Version @addArg)
                        if ($appInfo) {
                            $publishIt = $false
                            Write-Host "$($navAppInfo.Name) is already published"
                            if ($appInfo.IsInstalled) {
                                $install = $false
                                Write-Host "$($navAppInfo.Name) is already installed"
                               ########## new ##########
                                $upgrade = $false
                                Write-Host "$($navAppInfo.Name) is already upgraded"
                               #######################
                            }
                        }
 
```

Sorry I did not have time to create a pull request, but it seems to work like this. at leat together with the ignoreIfAppExists switch

Without ignoreIfAppExists upgrade fails as expected

I am not sure if I tested all possible scenarios

**Additional context**
- this happens all the time. Bccontainerhelper version 3.0.1?
- i never tried to use publish-navcontainerapp with both -install -upgrade before

**Comment by freddydk** [comment]
Sorry for the delay

**Comment by freddydk** [comment]
fixed in 3.0.4
